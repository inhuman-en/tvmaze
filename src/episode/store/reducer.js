import { createReducer } from '@reduxjs/toolkit';
import * as actions from './actionCreators';

const initialState = {
  data: null,
  error: null,
  isLoading: false,
};

const reducer = createReducer(initialState, (builder) => {
  builder
    .addCase(actions.loadEpisodeDetails, (state) => {
      state.isLoading = true;
      state.data = null;
      state.error = null;
    })
    .addCase(actions.loadEpisodeDetailsSuccess, (state, action) => {
      state.isLoading = false;
      state.data = action.payload;
    })
    .addCase(actions.loadEpisodeDetailsFail, (state, action) => {
      state.isLoading = false;
      state.error = action.error;
    });
});

export default reducer;
