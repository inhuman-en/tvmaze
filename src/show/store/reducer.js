import { combineReducers } from 'redux';
import { createReducer } from '@reduxjs/toolkit';
import * as actions from './actionCreators';

const initialDetailsState = {
  data: null,
  error: null,
  isLoading: false,
};

const initialEpisodeListState = {
  data: null,
  error: null,
  isLoading: false,
};

export const detailsReducer = createReducer(initialDetailsState, (builder) => {
  builder
    .addCase(actions.loadShowDetails, (state) => {
      state.isLoading = true;
      state.data = null;
      state.error = null;
    })
    .addCase(actions.loadShowDetailsSuccess, (state, action) => {
      state.isLoading = false;
      state.data = action.payload;
    })
    .addCase(actions.loadShowDetailsFail, (state, action) => {
      state.isLoading = false;
      state.error = action.error;
    });
});


export const episodeListReducer = createReducer(initialEpisodeListState, (builder) => {
  builder
    .addCase(actions.loadEpisodeList, state => {
      state.isLoading = true;
      state.data = null;
      state.error = null;
    })
    .addCase(actions.loadEpisodeListSuccess, (state, action) => {
      state.isLoading = false;
      state.data = action.payload;
    })
    .addCase(actions.loadEpisodeListFail, (state, action) => {
      state.isLoading = false;
      state.error = action.error;
    });
});

export default combineReducers({
  details: detailsReducer,
  episodeList: episodeListReducer
});
