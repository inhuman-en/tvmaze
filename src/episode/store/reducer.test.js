import reducer from './reducer';

describe('episode reducer', () => {
  const initialState = {
    data: null,
    error: null,
    isLoading: false,
  };

  it('should return correct state after [EPISODE] EPISODE_DETAILS_LOAD_REQUEST action', () => {
    const action = {
      type: '[EPISODE] EPISODE_DETAILS_LOAD_REQUEST',
      payload: 1,
    }
    const result = reducer(initialState, action);
    expect(result).toMatchSnapshot();
  });

  it('should return correct state after [EPISODE] EPISODE_DETAILS_LOAD_SUCCESS action', () => {
    const mockResponse = {
      id: 1,
      name: 'episode name',
      image: null,
      airDate: '2017-04-04',
    };
    const action = {
      type: '[EPISODE] EPISODE_DETAILS_LOAD_SUCCESS',
      payload: mockResponse,
    }
    const result = reducer(initialState, action);
    expect(result).toMatchSnapshot();
  });

  it('should return correct state after [EPISODE] EPISODE_DETAILS_LOAD_FAIL action', () => {
    const mockError = 'Something bad happened';
    const action = {
      type: '[EPISODE] EPISODE_DETAILS_LOAD_FAIL',
      error: mockError,
    }
    const result = reducer(initialState, action);
    expect(result).toMatchSnapshot();
  });
});
