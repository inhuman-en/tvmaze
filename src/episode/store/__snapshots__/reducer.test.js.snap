// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`episode reducer should return correct state after [EPISODE] EPISODE_DETAILS_LOAD_FAIL action 1`] = `
Object {
  "data": null,
  "error": "Something bad happened",
  "isLoading": false,
}
`;

exports[`episode reducer should return correct state after [EPISODE] EPISODE_DETAILS_LOAD_REQUEST action 1`] = `
Object {
  "data": null,
  "error": null,
  "isLoading": true,
}
`;

exports[`episode reducer should return correct state after [EPISODE] EPISODE_DETAILS_LOAD_SUCCESS action 1`] = `
Object {
  "data": Object {
    "airDate": "2017-04-04",
    "id": 1,
    "image": null,
    "name": "episode name",
  },
  "error": null,
  "isLoading": false,
}
`;
